contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -1298,14 +1331,19 @@ clean_up_bbf_server_def()
 {
 	char 			*query_str;
 	StringInfoData 	query;
+	int save_nestlevel;
 
 	initStringInfo(&query);
 
 	appendStringInfo(&query, "TRUNCATE TABLE sys.babelfish_server_options CASCADE");
 
 	query_str = query.data;
 
+	save_nestlevel = NewGUCNestLevel();
+	SetConfigOption("allow_system_table_mods", "on",
+					GUC_CONTEXT_CONFIG, PGC_S_SESSION);
 	exec_utility_cmd_helper(query_str);
```

Comment:
should we add try/catch to avoid failure case which can leak the option setting? also, why not re-set the old value but use nesting level to rollback?

Reply:
This was initially suggested by Kuntal in the above comment to use GUC nesting level instead of try-catch, do you not agree with that?

Reply:
I have changed the logic back to try-catch.